### ISSUE TITLE
Document build process

### ISSUE THREAD

**Issue by dirk** [bakkdoor/fancy#74]
Right now there's not much reference to go on for building besides just `rake`. For example I'm not sure what the right process is for rebuilding the Fancy runtime environment (stable Rbx then bootstrap, right?) without having to rebuild the whole thing (`rake clean` followed by `rake`).

**Comment by bakkdoor** [comment]
Yeah, we probably should add some documentation for that. Right now you can just run `rake` whenever you changed something and that will either succeed or fail. If it fails, it means you need to bootstrap again (and possibly fix a bug you introduced with that change that caused the breakage).
It's a bit annoying having to rebuild, I guess we could look into improving that, but there's always going to be similar bootstrapping issues with a self-hosted language like Fancy. Introducing a subtle bug can cause many problems since it's interdependent at many levels.

**Comment by bakkdoor** [comment]
But at least rebuilding & bootstrapping Fancy doesn't take _that_ long. There's systems that are way worse in that regard and that take way longer than just a couple of seconds. It currently takes me about 15 seconds to bootstrap Fancy from scratch, which I think is pretty good. :)
